### PR TITLE
Explosions can destroy unbreakable blocks like bedrock

### PR DESCRIPTION
**Pull request by SunsetTadpole78**
## Introduction
currently on pocketmine, explosions break unbreakable blocks like bedrock, this pull request allows to solve this problem or to force this breaking of blocks

## Changes
- patch of the bug which allows to break unbreakable blocks
- addition of a condition in the ExplosionPrimeEvent to be able to break the unbreakable blocks (allows to force the explosion)

## Tests
Before:
before
After:
after

### CODE REVIEW

File: src/event/entity/ExplosionPrimeEvent.php [hunk 1]
```diff
@@ -60,12 +61,23 @@ public function setForce(float $force) : void{
 		}
 		$this->force = $force;
 	}
-
```
Comment (ShockedPlot7560) [inline src/event/entity/ExplosionPrimeEvent.php]: It is an unnecessary change

File: src/world/Explosion.php [hunk 2]
```diff
@@ -129,6 +129,10 @@ public function explodeA() : bool{
 									if(!isset($this->affectedBlocks[World::blockHash($vBlockX, $vBlockY, $vBlockZ)])){
 										$_block = $this->world->getBlockAt($vBlockX, $vBlockY, $vBlockZ, true, false);
 										foreach($_block->getAffectedBlocks() as $_affectedBlock){
+
+											if(!$canBreakAll && !$_affectedBlock->getBreakInfo()->isBreakable())
```
Comment (ShockedPlot7560) [inline src/world/Explosion.php]: Use the block form instead of the inline ?
```php
if(...){
     ...
}
```

File: src/event/entity/ExplosionPrimeEvent.php [hunk 3]
```diff
@@ -38,16 +38,17 @@
 class ExplosionPrimeEvent extends EntityEvent implements Cancellable{
 	use CancellableTrait;
 
-	/** @var float */
-	protected $force;
+	protected float $force;
```
Comment (ShockedPlot7560) [inline src/event/entity/ExplosionPrimeEvent.php]: This is an unrelated change

**Comment by dktapps** [comment]
Bedrock isn't indestructible. It just has a very (very) high blast resistance (18,000,000). The point is that small amounts of explosive can't destroy it, but a powerful enough explosive might be able to.

TNT in PM isn't able to destroy bedrock, so I don't see why this PR is necessary.
